functions/param-names.js

```diff
@@ -14,6 +14,16 @@ module.exports = (targetVal, _opts, paths) => {
     return [];
   }
 
+  if (['path', 'query'].includes(targetVal.in) && targetVal.name !== 'api-version') {
```

Comment:
You don't want this line.  Because of it the check for '$' and "@' are skipped for header parameters.  Delete this and line 26 (the closing brace).